bee-message/src/payload/milestone/option/mod.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+mod pow;
+mod receipt;
+
+use derive_more::{Deref, From};
+use iterator_sorted::is_unique_sorted;
+use packable::{bounded::BoundedU8, prefix::BoxedSlicePrefix, Packable};
+pub(crate) use receipt::{MigratedFundsAmount, ReceiptFundsCount};
+
+pub use self::{
+    pow::PowMilestoneOption,
+    receipt::{MigratedFundsEntry, ReceiptMilestoneOption, TailTransactionHash},
+};
+use crate::Error;
+
+///
+#[derive(Clone, Debug, Eq, PartialEq, From, Packable)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(tag = "type", content = "data")
+)]
+#[packable(unpack_error = Error)]
+#[packable(tag_type = u8, with_error = Error::InvalidMilestoneOptionKind)]
+pub enum MilestoneOption {
+    /// A receipt milestone option.
+    #[packable(tag = ReceiptMilestoneOption::KIND)]
+    Receipt(ReceiptMilestoneOption),
+    /// A PoW milestone option.
+    #[packable(tag = PowMilestoneOption::KIND)]
+    Pow(PowMilestoneOption),
+}
+
+impl MilestoneOption {
+    /// Return the milestone option kind of a [`MilestoneOption`].
+    pub fn kind(&self) -> u8 {
+        match self {
+            Self::Receipt(_) => ReceiptMilestoneOption::KIND,
+            Self::Pow(_) => PowMilestoneOption::KIND,
+        }
+    }
+}
+
+pub(crate) type MilestoneOptionCount = BoundedU8<0, { MilestoneOptions::COUNT_MAX }>;
+
+///
```

Comment:
Docs

Reply:
You don't have to comment on them all as I won't do it in this PR and there are a lot of them ahah